### ISSUE TITLE
Adding an optional cosine scheduler

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/lit-gpt#406]
We might want to add a cosine scheduler and enable it via a command-line arg. It's pretty much standard for LLM pretraining an finetuning.

**Comment by patrickhwood** [comment]
It's used in the openwebtext and redpajama pretrain scripts, but nowhere else.  If I have time this week I make take a quick look at whether it improves the lora fine tuning (which seems to bounce around a lot once the loss falls to around 1.
